code/headers/mp3_player.hpp

```diff
@@ -0,0 +1,170 @@
+#pragma once
+
+#include "sound.hpp"
+
+#include <array>
+#include <hwlib.hpp>
+
+namespace r2d2::sound {
+    /**
+     * Class mp3_player_c plays mp3 files.
+     */
+    class mp3_player_c : public sound_c {
+    private:
+        hwlib::pin_out &power_pin;
+        hwlib::pin_oc &volume_up_next_pin;
+        hwlib::pin_oc &volume_down_prev_pin;
+        hwlib::pin_oc &play_pause_mode_pin;
+        hwlib::pin_oc &repeat_pin;
+
+        /**
+         * This function initializes the mp3 player.
+         *
+         * @internal
+         */
+        void init();
+
+        /**
+         * This function simulates a button press for a given amount of
+         * milliseconds.
+         *
+         * @internal
+         * @param milliseconds The amount of milliseconds you want the pin to be
+         * pressed.
+         * @param pin The pin used for the press simulation.
+         */
+        void press_for(unsigned int milliseconds, hwlib::pin_oc &pin);
+
+        /**
+         * This function simulates a 2 button press simultaniusly for a given
+         * amount of milliseconds.
+         *
+         * @internal
+         * @param milliseconds The amount of milliseconds you want the pins to
+         * be pressed.
+         * @param 2 pins used for the press simulation.
+         */
+        void press_for(unsigned int milliseconds, hwlib::pin_oc &pin1,
+                       hwlib::pin_oc &pin2);
+
+        /**
+         * This function uses the press_for function to output a short press
+         * command.
+         *
+         * @internal
+         * @param pin The pin you want to use for the short press.
+         */
+        void short_press(hwlib::pin_oc &pin);
```

Comment:
Misschien ligt het aan mij maar ik snap niet helemaal wat het idee is met het simuleren van button presses... Heeft dit daadwerkelijk te maken met de mp3 player? Wat zijn de pinnen die je meegeeft?